src/arithmetic_circuit.rs

```diff
@@ -0,0 +1,180 @@
+use std::{
```

Comment:
This is basically a copy of the compiler's `ArithmeticCircuit` (project-specific errors replaced with just `String` for now). We need to replace this with `ArithmeticCircuit` from an independent crate.